internal/filter/filter/filter.go

```diff
@@ -0,0 +1,11 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package filter // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filter"
+
+// Filter is an interface for matching values against a set of filters.
+type Filter interface {
```

Comment:
`internal/filter` is already filled with interfaces (many which I'd like to remove). Do we definitely need to add another? Nothing in `expr` can be reused?

Reply:
I guess it is ok bc it is internal and we can easily update later.  Can we name it something more related to mdatagen/scrappers/config? `internal/filter/filter` isn't very descriptive.

Reply:
Also, filterset converts other types to string, can we do that?  What requires the use of `any` if we are doing regex/strict string matching?

Reply:
> I think the general idea is to have this filtering interface as the standard interface on receivers (not just mdatagen). It basically implements the Option B of this issue https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/25161

You're right, I forgot this was part of that issue.